integration_tests/README.md

```diff
@@ -303,3 +303,131 @@ The marks you care about are all in marks.py
+at query compile time, before the Rapids Accelerator is invoked. Thus, adding multiple combinations of literal inputs
+need not necessarily add more test coverage.
+
+### 4. Null values
+Ensure that the test data accommodates null values for input columns. This includes null values in columns
+and in literal inputs.
+
+Null values in input columns are a frequent source of bugs in the Rapids Accelerator for Spark, 
+because of mismatches in null-handling and semantics, between RAPIDS `libcudf` (on which 
+the Rapids Accelerator relies heavily), and Apache Spark. 
+
+Tests for aggregations (including group-by, reductions, and window aggregations) should cover cases where 
+some rows are null, and where *all* input rows are null.
+
+Apart from null rows in columns of primitive types, the following conditions must be covered for nested types:
+
+   * Null rows at the "top" level for `Array`/`Struct` columns.   E.g. `[ [1,2], [3], ∅, [4,5,6] ]`.
+   * Non-null rows containing null elements in the child column. E.g. `[ [1,2], [3,∅], ∅, [4,∅,6] ]`.
+   * All null rows at a nested level. E.g. 
+     * All null list rows: `[ ∅, ∅, ∅, ∅ ]`
+     * All null elements within list rows: `[ [∅,∅], [∅,∅], [∅,∅], [∅,∅] ]`
+
+The `DataGen` classes in `integration_tests/src/main/python/data_gen.py` can be configured to generate null values
+for the cases mentioned above.
+
+### 5. Empty rows in `Array` columns
+Operations on `ArrayType` columns must be tested with input columns containing non-null *empty* rows.
+E.g.
+```
+[
+    [0,1,2,3],
+    [], <------- Empty, non-null row.
+    [4,5,6,7],
+    ...
+]
+```
+Using the `ArrayGen` data generator in `integration_tests/src/main/python/data_gen.py` will generate
+empty rows as mentioned above.
+
+### 6. Degenerate cases with "empty" inputs
+Ensure that operations are tested with "empty" input columns (i.e. containing zero rows.)
+
+E.g. `COUNT()` on an empty input column yields `0`. `SUM()` yields `0` for the appropriate numeric type.
+
+### 7. Special floating point values
+Apart from `null` values, `FloatType` and `DoubleType` input columns must also include the following special values:
+   * +/- Zero
+   * +/- Infinity
+   * +/- NaN
+
+Note that the special values for floating point numbers might have different bit representations for the same
+equivalent values. The [Java documentation for longBitsToDouble()](https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html#longBitsToDouble-long-)
+describes this with examples. Operations should be tested with multiple bit-representations for these special values.
+
+The `FloatGen` and `DoubleGen` data generators in `integration_tests/src/main/python/data_gen.py` can be configured
+to generate the special float/double values mentioned above.
+
+Note that floating point values generated on the GPU might not match those from the CPU exactly. Any differences will
```

Comment:
This makes it sound like the GPU for arbitrary operations will get the wrong floating point values.  I think we should be clear that for most operations this will not happen and point to https://github.com/NVIDIA/spark-rapids/blob/branch-21.08/docs/compatibility.md#floating-point for more information about when to use `@approximate_float` and when not to.

Reply:
I should've known better than to editorialize. I've corrected the wording, and pointed to `compatibility.md`.